pkg/state/state_ofed.go

```diff
@@ -78,21 +96,94 @@ type stateOFED struct {
 	stateSkel
 }
 
+type additionalVolumeMounts struct {
+	VolumeMounts []v1.VolumeMount
+	Volumes      []v1.Volume
+}
+
 type ofedRuntimeSpec struct {
 	runtimeSpec
 	CPUArch        string
 	OSName         string
 	OSVer          string
 	MOFEDImageName string
-	HTTPProxy      string
-	HTTPSProxy     string
-	NoProxy        string
 }
 
 type ofedManifestRenderData struct {
-	CrSpec       *mellanoxv1alpha1.OFEDDriverSpec
-	NodeAffinity *v1.NodeAffinity
-	RuntimeSpec  *ofedRuntimeSpec
+	CrSpec                 *mellanoxv1alpha1.OFEDDriverSpec
+	NodeAffinity           *v1.NodeAffinity
+	RuntimeSpec            *ofedRuntimeSpec
+	AdditionalVolumeMounts additionalVolumeMounts
+}
+
+// getCertConfigPath returns the standard OS specific path for ssl keys/certificates
+func getCertConfigPath(osname string) (string, error) {
+	if path, ok := CertConfigPathMap[osname]; ok {
+		return path, nil
+	}
+	return "", fmt.Errorf("distribution not supported")
+}
+
+// getRepoConfigPath returns the standard OS specific path for repository configuration files
+func getRepoConfigPath(osname string) (string, error) {
+	if path, ok := RepoConfigPathMap[osname]; ok {
+		return path, nil
+	}
+	return "", fmt.Errorf("distribution not supported")
+}
+
+// FromConfigMap generates Volumes and VolumeMounts data for the specified ConfigMap object
+func (a *additionalVolumeMounts) FromConfigMap(configMap *v1.ConfigMap, destDir string) error {
+	volumeMounts, itemsToInclude, err := a.createConfigMapVolumeMounts(configMap, destDir)
+	if err != nil {
+		return fmt.Errorf("ERROR: failed to create ConfigMap VolumeMounts for custom repo config: %v", err)
```

Comment:
i think its clear its an error so no need for ERROR: prefix.

also error should not reference custom repo config but rather something generic like:
"failed to create Volume mounts from configmap. %v, err"